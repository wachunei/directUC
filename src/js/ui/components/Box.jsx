import styled, { css } from "styled-components";

const Box = styled.div`
  display: flex;
  flex-direction: ${({ horizontal }) => (horizontal ? "row" : "column")};

  ${({ horizontal, center, start, end, between, around }) => {
    if (!horizontal) {
      if (start) {
        return css`
          align-items: flex-start;
        `;
      }
      if (center) {
        return css`
          align-items: center;
        `;
      }
      if (end) {
        return css`
          align-items: flex-end;
        `;
      }
      return null;
    }

    if (start) {
      return css`
        justify-content: flex-start;
      `;
    }
    if (center) {
      return css`
        justify-content: center;
      `;
    }
    if (end) {
      return css`
        justify-content: flex-end;
      `;
    }
    if (between) {
      return css`
        justify-content: space-between;
      `;
    }
    if (around) {
      return css`
        justify-content: space-around;
      `;
    }
    return null;
  }}

  ${({ horizontal, top, middle, bottom }) => {
    if (horizontal) {
      if (top) {
        return css`
          align-items: flex-start;
        `;
      }
      if (middle) {
        return css`
          align-items: center;
        `;
      }
      if (bottom) {
        return css`
          align-items: flex-end;
        `;
      }
      return null;
    }
    return null;
  }}
`;

export default Box;
