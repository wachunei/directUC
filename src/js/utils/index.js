import browser from "webextension-polyfill";

export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = process.env.TARGET === "firefox";

export const trackingId =
  process.env.NODE_ENV === "development" ? "UA-177532719-1" : "UA-62971405-1";

export const { version } = browser.runtime.getManifest();
