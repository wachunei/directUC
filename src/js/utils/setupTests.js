// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
