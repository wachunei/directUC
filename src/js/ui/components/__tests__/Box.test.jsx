import Box from "../Box";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Brand Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Box />);
    expect(tree).toMatchSnapshot();
  });

  test("should match snapshot with light theme", () => {
    const tree = renderTree(<Box />, "lightTheme");
    expect(tree).toMatchSnapshot();
  });

  test("should match snapshot with dark theme", () => {
    const tree = renderTree(<Box />, "darkTheme");
    expect(tree).toMatchSnapshot();
  });

  test("should match with props horizontal", () => {
    const tree = renderTree(<Box horizontal />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal top", () => {
    const tree = renderTree(<Box horizontal top />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal middle", () => {
    const tree = renderTree(<Box horizontal middle />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal bottom", () => {
    const tree = renderTree(<Box horizontal bottom />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal start", () => {
    const tree = renderTree(<Box horizontal start />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal between", () => {
    const tree = renderTree(<Box horizontal between />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal around", () => {
    const tree = renderTree(<Box horizontal around />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props horizontal end", () => {
    const tree = renderTree(<Box horizontal end />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props start", () => {
    const tree = renderTree(<Box start />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props center", () => {
    const tree = renderTree(<Box center />);
    expect(tree).toMatchSnapshot();
  });
  test("should match with props end", () => {
    const tree = renderTree(<Box end />);
    expect(tree).toMatchSnapshot();
  });
});
