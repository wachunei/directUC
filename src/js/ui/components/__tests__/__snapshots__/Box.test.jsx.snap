// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand Component should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match snapshot with dark theme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match snapshot with light theme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props end 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  className="c0"
  end={true}
/>
`;

exports[`Brand Component should match with props horizontal 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props horizontal around 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props horizontal between 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props horizontal bottom 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props horizontal end 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
  end={true}
/>
`;

exports[`Brand Component should match with props horizontal middle 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props horizontal start 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
  start={true}
/>
`;

exports[`Brand Component should match with props horizontal top 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`Brand Component should match with props start 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
  start={true}
/>
`;
